server/plugin/api_test.go

```diff
@@ -347,3 +347,785 @@ func TestSearchRecordsInServiceNow(t *testing.T) {
+			RequestBody: "{}",
+			SetupAPI: func(api *plugintest.API) *plugintest.API {
+				api.On("LogError", testutils.GetMockArgumentsWithType("string", 3)...).Return()
+				return api
+			},
+			SetupClient: func(client *mock_plugin.Client) {
+				client.On("CheckForDuplicateSubscription", mock.AnythingOfType("*serializer.SubscriptionPayload")).Return(
+					false, http.StatusOK, nil,
+				)
+
+				client.On("CreateSubscription", mock.AnythingOfType("*serializer.SubscriptionPayload")).Return(
+					http.StatusForbidden, fmt.Errorf("create subscription error"),
+				)
+
+				var s *serializer.SubscriptionPayload
+				monkey.PatchInstanceMethod(reflect.TypeOf(s), "IsValidForCreation", func(_ *serializer.SubscriptionPayload, _ string) error {
+					return nil
+				})
+			},
+			ExpectedStatusCode:   http.StatusForbidden,
+			ExpectedErrorMessage: "create subscription error",
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			assert := assert.New(t)
+			api := test.SetupAPI(&plugintest.API{})
+			defer api.AssertExpectations(t)
+			defer monkey.UnpatchAll()
+
+			p := setupTestPlugin(api, nil)
+			client := setupPluginForSubscriptionsConfiguredMiddleware(p, t)
+			test.SetupClient(client)
+
+			w := httptest.NewRecorder()
+			r := httptest.NewRequest(requestMethod, requestURL, bytes.NewBufferString(test.RequestBody))
+			r.Header.Add(constants.HeaderMattermostUserID, testutils.GetID())
+			p.ServeHTTP(nil, w, r)
+
+			result := w.Result()
+			require.NotNil(t, result)
+			defer result.Body.Close()
+
+			assert.Equal(test.ExpectedStatusCode, result.StatusCode)
+			if test.ExpectedErrorMessage != "" {
+				var resp *serializer.APIErrorResponse
+				err := json.NewDecoder(result.Body).Decode(&resp)
+				require.Nil(t, err)
+
+				assert.Contains(resp.Message, test.ExpectedErrorMessage)
+			}
+		})
+	}
+}
+
+func TestGetAllSubscriptions(t *testing.T) {
+	requestURL := fmt.Sprintf("%s%s", constants.PathPrefix, constants.PathGetAllSubscriptions)
+	requestMethod := http.MethodGet
+	limit, offset := testutils.GetLimitAndOffset()
+	for name, test := range map[string]struct {
+		ChannelID            string
+		UserID               string
+		SubscriptionType     string
+		SetupAPI             func(*plugintest.API) *plugintest.API
+		SetupClient          func(client *mock_plugin.Client)
+		ExpectedStatusCode   int
+		ExpectedErrorMessage string
+		ExpectedCount        int
+	}{
+		"success": {
+			SetupAPI: func(api *plugintest.API) *plugintest.API {
+				return api
+			},
+			SetupClient: func(client *mock_plugin.Client) {
+				client.On("GetAllSubscriptions", "", "", "", limit, offset).Return(
+					testutils.GetSubscriptions(4), http.StatusOK, nil,
+				)
+
+				client.On("GetRecordFromServiceNow", constants.SubscriptionRecordTypeProblem, "").Return(
+					testutils.GetServiceNowRecord(), http.StatusOK, nil,
+				)
+			},
+			ExpectedStatusCode: http.StatusOK,
+			ExpectedCount:      4,
+		},
+		"invalid channel id": {
+			ChannelID: "dsf",
```

Comment:
usually, we should take mock values similar to the actual variable. Check this everywhere
```suggestion
			ChannelID: "test-channelID",
```